sdk/src/store.rs

```diff
@@ -3988,51 +4031,26 @@ impl Store {
 
     /// Handles remote manifests asynchronously when fetch_remote_manifests feature is enabled
     /// Required because wasm-bindgen cannot use async functions in a sync context.
-    #[cfg(all(target_arch = "wasm32", not(target_os = "wasi")))]
+    #[cfg(target_arch = "wasm32")]
     async fn handle_remote_manifest_async(ext_ref: &str) -> Result<Vec<u8>> {
```

Comment:
Isn't this already supported for non-wasm32 as well?

Reply:
I'm combining both wasm-bindgen and WASI into the same function.

Reply:
Yes. We can discuss and raise a second PR if we want a non-wasm async version.